velox/expression/tests/ArgumentTypeFuzzer.cpp

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/expression/tests/ArgumentTypeFuzzer.h"
+
+#include <boost/algorithm/string.hpp>
+#include <velox/type/Type.h>
+
+namespace facebook::velox::test {
+
+namespace {
+
+// Return a random type among those in kSupportedTypes determined by seed.
+// todo: Extend this function to return arbitrary random types including nested
+// complex types.
+TypePtr randomType(std::mt19937& seed) {
+  static std::vector<TypePtr> kSupportedTypes{
+      BOOLEAN(),
+      TINYINT(),
+      SMALLINT(),
+      INTEGER(),
+      BIGINT(),
+      REAL(),
+      DOUBLE(),
+      TIMESTAMP(),
+      DATE(),
+      INTERVAL_DAY_TIME(),
+      SHORT_DECIMAL(10, 5),
+      LONG_DECIMAL(20, 10)};
+  auto index = folly::Random::rand32(kSupportedTypes.size(), seed);
+  return kSupportedTypes[index];
+}
+
+std::string fromTypeToBaseName(const TypePtr& type) {
+  return boost::algorithm::to_lower_copy(std::string{type->kindName()});
+}
+
+std::optional<TypeKind> fromBaseNameToTypeKind(std::string& typeName) {
+  auto kindName = boost::algorithm::to_upper_copy(typeName);
+  return tryMapNameToTypeKind(kindName);
+}
+
+} // namespace
+
+TypePtr ArgumentTypeFuzzer::bindPossibleType(
+    const exec::TypeSignature& formalType) {
+  // Not supporting argument types that contain "any".
+  std::string type = formalType.baseType();
+  if (type == "any") {
+    VELOX_NYI();
+  }
+
+  auto it = bindings_.find(type);
+  if (it != bindings_.end()) {
+    if (!it->second) {
+      it->second = randomType(seed_);
+    }
+    VELOX_DCHECK(it->second);
+    return it->second;
+  } else {
+    auto typeKind = fromBaseNameToTypeKind(type);
+    if (!typeKind.has_value()) {
```

Comment:
When would this happen?

Reply:
When the base name of a type signature is not the lower-case version of a type kind name (e.g., base name "array" maps to kind name "ARRAY"; base name "any" does not map to any kind name). fromBaseNameToTypeKind() uses tryMapNameToTypeKind() in Type.h to do this mapping.